UNIXFS.md

```diff
@@ -59,6 +64,27 @@ For files that are comprised of more than a single block, the 'Type' field will
 
 This data is serialized and placed inside the 'Data' field of the outer merkledag protobuf, which also contains the actual links to the child nodes of this object.
 
+## Metadata
+
+There are three ways to specify file metadata:
+
+1. The repeated `metadata` field in a directory applies metadata to each file in the directory.
+2. An intermediary node with a `Type` of `Metadata` applies metadata to an individual file. While this feature should be supported, it has been deprecated.
+3. An optional `defaultMetadata` field to specify the _default_ metadata for files in the directory. If unspecified, the default mode is 0755 and the default modification time is the epoch.
+  * Files: The default metadata is applied as-is.
+  * Directories:
+    * The default mode for directories is `defaultMetadata.mode | 0111` (sets the execute bit).
+    * The default mime type for directories is `inode/directory`.
+  * Symlinks:
+    * The default mime type for symlinks is `inode/symlink`.
+    * The mode for symlinks is _always_ 0777.
+
+Fields:
+
+* `mimeType` -- The mime-type of the file. This generally shouldn't be used.
```

Comment:
> This generally shouldn't be used.

Please can we document why? We wouldn't have to do sniffing in many cases if it's explicit.